src/main/java/com/epam/aidial/core/controller/InvitationController.java

```diff
@@ -40,14 +44,20 @@ public Future<?> getInvitations() {
     }
 
     public Future<?> getOrAcceptInvitation(String invitationId) {
-        String accept = context.getRequest().getParam("accept");
-        if (accept != null) {
+        boolean accept = Boolean.parseBoolean(context.getRequest().getParam("accept"));
+        if (accept) {
             proxy.getVertx()
                     .executeBlocking(() -> {
                         String bucketLocation = BlobStorageUtil.buildInitiatorBucket(context);
                         String bucket = encryptionService.encrypt(bucketLocation);
-                        shareService.acceptSharedResources(bucket, bucketLocation, invitationId);
-                        return null;
+                        ResourceDescription invitationResource = invitationService.getInvitationResource(invitationId);
+                        if (invitationResource == null) {
+                            throw new ResourceNotFoundException();
+                        }
+                        return lockService.underUserLock(proxy, context, () -> {
+                            shareService.acceptSharedResources(bucket, bucketLocation, invitationId);
```

Comment:
It uses two buckets: my (target) and a different user (source). So you need to lock source user bucket to defend against concurrent deletion of resources in source user's invitations.